src/script/lua_api/l_modchannels.cpp

```diff
@@ -33,9 +33,8 @@ int ModApiChannels::l_mod_channel_join(lua_State *L)
 		return 0;
 
 	getGameDef(L)->joinModChannel(channel);
-	ModChannel *channelObj = getGameDef(L)->getModChannel(channel);
-	assert(channelObj);
-	ModChannelRef::create(L, channelObj);
+	assert(getGameDef(L)->isModChannelRegistered(channel));
```

Comment:
`assert(getGameDef(L)->getModChannel(channel));` would be possible here too because `isModChannelRegistered` is used nowhere else.

Reply:
yeah it would be possible, just seemed cleaner at the moment to do it this way.